### DISCUSSION TITLE
Statistics recording layout

### DISCUSSION THREAD

**Discussion by rgmf** [OpenTracksApp/OpenTracks#813] [Ideas]
I'm riding my bike a lot and I put my phone on the handlebar so I can see speed, time, gain, etc. But somehow I don't like how speed and data is presented on the screen, mainly because speed is smaller than time, for example, but speed is the field I see the more.

TL;DR after check several options I'm using this layout and I like it:
![imagen](https://user-images.githubusercontent.com/4819612/121399988-51676380-c957-11eb-9a9c-f70b9d5f952a.png)

The changes I've done are the following:
- Total time have been removed (you can see it on fragment controller and the important one is total moving time).
- Activity icon have been removed (it's not a field I need when I'm recording).
- Numbers are more important than labels so numbers are big and the labels are small and below numbers. I've put the focus on the numbers.
- All fields have the same size.

@dennisguse, @pstorch, and everyone reading this: what do you think? Don't you think we should improve this screen? I think so now that I'm riding my bike (when I run the smartphone is in my pocket and I don't mind). I also think we should do something similar we can see in cyclocomputers.

**Comment by dennisguse** [comment]
Love it! Especially the common font size!

# Data
My opinion about the data to show:

While using a race bike (also on the handlebar), I use:
* Speed (quite often)
* Heartrate (sometimes)
* Avg speed
* Distance 
* (assumption) cadence and power meter if people have such sensors)

Sometimes (mainly for unknown tracks aka not the common rides):
* Gain/loss

Don't need that (at least on not in the top part):
* Max. speed (almost never; really don't care while cycling; nice to know afterwards)
* elevation (only for hiking)

Never used:
* lat/lng

PS: I live in an area without any mountains. 50km cycling gives me 200m gain/loss.
That's why I normally don't need this data - only while traveling (usually abroad).

PPS: For landscape mode, we could also use three columns. i can attach my phone also in horizontal position onto the handlebar :)

**Reply by dennisguse** [reply]
On a dedicated device, a friend of mine is using the following layout (for race cycling).
He is training based upon power and cadence.

![tacho](https://user-images.githubusercontent.com/2820825/121797303-bce35680-cc1f-11eb-8b2a-dc5161723a67.jpg)

**Reply by dennisguse** [reply]
And this one is somehow related: https://github.com/OpenTracksApp/OpenTracks/issues/302
Would be great to see if the sensor or the GPS is used.

**Reply by rgmf** [reply]
> On a dedicated device, a friend of mine is using the following layout (for race cycling).
> He is training based upon power and cadence.
> 
> ![tacho](https://user-images.githubusercontent.com/2820825/121797303-bce35680-cc1f-11eb-8b2a-dc5161723a67.jpg)

I'm thinking on how to do a custom data screen. I'll have some more time on July and I'd like to work on it. I'll do a detailed proposal in the next days.

Why to do it? Because everybody likes/needs different screen's data and I think it would a great feature in OT.

**Reply by dennisguse** [reply]
That would be perfect!
If you need anything, let me know!

PS: My plan for July is to implement sensor-only recording (aka no GPS use).
Using a cycling sensor is actually quite power efficient ;)

**Reply by oriolgallemi** [reply]
Re custom or tailored screenshots, there are 2 user profiles.

Amateurs, requiring real time data on screen.
- a track map
- altitude
- heartrate
- stopwatch 
- plenty of data (average speed, max speed, distance, part distance, time, coordinates lat/long/alt, slope, max slope, min slope, satellite visibility (count), gain+, gain-, total elevation)
- silly indicators (calories, date, moving time, total time, average moving speed,)
- Graphs and plots of all sorts
- never require export other than track coordinates (lat/long/alt) for sharing

Personally, I don't like having a wealth of data as I prefer meaningful instant information to digest it immediately.

**Sportspersons**: more focused on small real time data and heavily depending on sport (i.e trekking in low visibility, bearing and track is paramount, coordinates only OK in case of emergency/sailing/climbing)
- map route or odometer
- milestone/time/distance announcements
- bearing
- speed km/h - min/km
- clock / GPS clock / elapsed time 
- "instant" cadence, heartrate, or power (averaged 20-30 seconds). One indicator is enough at a time
- export features for analytics (i.e. exporting in CSV) with all data including duplicates (i.e GPS speed Vs wheel rpm for bias correction)
- Dark mode / black screen for the maximum of time to enhance battery span

Cool to have (I'll place it to :ideas)
- self-competition: recorded track Vs current track (time/distance gap in a traffic light arrow, from red (lag, behind) to green (lead, ahead)
- post-activity analytics (power, slope, speed, pace/cadence) It will consume far less energy than real time computing and plotting). I prefer a script/macro over a computer rather than on the phone
- Indicator on optimum speed (I am working on this, but requires individuals to parametrize themselves or use AI methods drawing processing power)

**Comment by edern-appere** [comment]
It's funny because yesterday, after riding my bike with my phone on the handlebar like the two of you, I sent an email to Dennis to ask for the exact same feature ! I sent an email because I'm noy used to use github but now I found this conversation thanks to Dennis.

So, if you can design something that suits better cyclist needs, it would be great. The simplest solution would be, I guess, to have one recording screen designed for any activity (ie the one currently in use in the app) and another one specially for cycling. But if you plan to make it customizable, that's the cherry on the cake !!

The information I use most while riding are :
- current speed
- Distance
- Avg speed
- Heartrate

**Reply by rgmf** [reply]
@edern-appere thank you for your opinion, that's really useful.

I think customization feature is the best option too, because everyone likes a custom  layout for biking. So let's going to try to do it.
